### PR TITLE
Add configurable rating delay for overlay flashcards

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

Adds a configurable rating delay feature that locks difficulty rating buttons for a user-specified duration after revealing a flashcard answer. This encourages users to spend time reviewing the card before submitting their rating.

The delay is configurable in settings (0–60 seconds), persisted via DataStore, and displayed as a countdown timer in the overlay. The feature gracefully handles edge cases: zero delay disables the lock, hiding the answer cancels the countdown, and re-revealing restarts it from the configured value.

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Infrastructure / CI/CD
- [ ] Performance improvement

## Changes Made

- **OverlayViewModel**: Added rating delay countdown logic triggered on answer reveal; cancels/restarts on hide/re-reveal; blocks rating submissions while locked
- **OverlayUiState**: Added `ratingDelaySeconds` and `ratingLockSecondsRemaining` fields
- **LearningCard**: Displays countdown timer and disables rating buttons while lock is active
- **SettingsViewModel & SettingsScreen**: Added rating delay setting with 0–60 second range validation
- **RatingDelaySettingsItem**: New composable for the settings UI row
- **DayCountersStore**: Persists rating delay preference; clamps values to [0, 60]
- **LearningPreferencesConfig**: Added `ratingDelaySeconds` field
- **Strings**: Added localized UI text for all supported languages (English, Russian, Ukrainian, Spanish, French, Italian, Chinese)

## How to Test

1. Open Settings and navigate to the rating delay option
2. Set a delay value (e.g., 5 seconds)
3. Open the overlay and reveal a flashcard answer
4. Verify the countdown timer appears and counts down to zero
5. Confirm rating buttons are disabled during the countdown
6. Hide and re-reveal the answer; verify the countdown restarts from the configured value
7. Wait for the countdown to reach zero and verify rating buttons become enabled
8. Test edge cases: zero delay (no countdown), negative/over-60 values (clamped in settings)
9. Verify the setting persists across app restarts

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases
  - Added 11 comprehensive unit tests in `OverlayViewModelTest` covering countdown behavior, locking/unlocking, cancellation, and error handling
  - Added UI tests in `OverlayScreenTest` verifying countdown display and button state
  - Added component test for `RatingDelaySettingsItem`
  - Updated existing tests in `DayCountersStoreTest`, `SettingsViewModelTest`, and `SettingsContentTest`
- [x] Existing tests pass locally
- [x] Updated documentation (strings) for all supported languages
- [x] No duplication introduced
- [x] No new warnings or suppressions introduced

## Related Issues

N/A

https://claude.ai/code/session_01Kyzmv3JALi7ewAakBFbt9L

### CODE REVIEW

File: app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt [hunk 1]
```diff
@@ -32,16 +32,19 @@ class DayCountersStore
...
         private companion object {
             const val DEFAULT_NEW_PER_DAY = 15
             const val DEFAULT_REVIEW_PER_DAY = 99
             const val DEFAULT_OVERLAY_INTERVAL_TIME = 0
+            const val DEFAULT_RATING_DELAY_SECONDS = 0
             const val MIN_LIMIT = 0
             const val MAX_NEW_PER_DAY = 200
             const val MAX_REVIEW_PER_DAY = 2000
             const val MAX_OVERLAY_INTERVAL_MINUTES = 2000
+            const val MAX_RATING_DELAY_SECONDS = 60
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt]: let's not store it in preferences store. Just a variable in actual settings modal should be enough

File: app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt [hunk 2]
```diff
@@ -142,4 +146,8 @@ class DayCountersStore
         suspend fun setOverlayInterval(value: Int) {
             ds.edit { it[K.OVERLAY_INTERVAL_TIME] = value.coerceIn(MIN_LIMIT, MAX_OVERLAY_INTERVAL_MINUTES) }
         }
+
+        suspend fun setRatingDelaySeconds(value: Int) {
+            ds.edit { it[K.RATING_DELAY_SECONDS] = value.coerceIn(MIN_LIMIT, MAX_RATING_DELAY_SECONDS) }
+        }
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt:142]: this can be done as validation in settings input

File: app/src/main/java/com/procrastilearn/app/ui/dojo/DojoScreen.kt [hunk 3]
```diff
@@ -137,6 +137,8 @@ internal fun DojoScreen(
                                         state = overlayState,
                                         onToggleShowAnswer = onToggleShowAnswer,
                                         onDifficultySelected = onDifficultySelected,
+                                        // Dojo is voluntary practice, not a gate: never delay rating.
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/ui/dojo/DojoScreen.kt]: remove comment

File: app/src/main/java/com/procrastilearn/app/ui/screens/settings/SettingsScreen.kt [hunk 4]
```diff
@@ -377,26 +393,86 @@ internal fun SettingsContent(
...
+    targetLanguage: Language,
+    onMixModeChange: (MixMode) -> Unit,
+    onStudyDirectionModeChange: (StudyDirectionMode) -> Unit,
+    onAddCardsForToday: (Int) -> Unit,
+    onNewPerDayChange: (Int) -> Unit,
+    onReviewPerDayChange: (Int) -> Unit,
+    onOverlayIntervalChange: (Int) -> Unit,
+    onRatingDelayChange: (Int) -> Unit,
+    onOpenAiApiKeyChange: (String) -> Unit,
+    onOpenAiPromptChange: (String) -> Unit,
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/ui/screens/settings/SettingsScreen.kt]: dont make fixes yet, let;s discuss in chat, is it possible to fix? maybe we can combine related props into objects together to reduce number of counts
